File: web/src/main/java/no/nav/fo/mia/config/CacheConfig.java
package no.nav.fo.mia.config;

import no.nav.fo.mia.utils.CacheKeyGenerator;
import org.springframework.cache.CacheManager;
import org.springframework.cache.annotation.EnableCaching;
import org.springframework.cache.jcache.JCacheCacheManager;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.cache.Caching;
import javax.cache.spi.CachingProvider;
import java.net.URISyntaxException;


@Configuration
@EnableCaching
public class CacheConfig {
    @Bean
    public CacheManager cacheManager() throws URISyntaxException {
        CachingProvider cachingProvider = Caching.getCachingProvider();
        javax.cache.CacheManager cacheManager = cachingProvider.getCacheManager(
                getClass().getResource("/cache/ehcache.xml").toURI(),
                cachingProvider.getDefaultClassLoader());

        return new JCacheCacheManager(cacheManager);
    }

    @Bean
    public CacheKeyGenerator cacheKeyGenerator() {
        return new CacheKeyGenerator();
    }
}

File: src/main/kotlin/no/nav/fo/mia/config/CacheConfig.kt
package no.nav.fo.mia.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.cache.CacheManager
import org.springframework.cache.annotation.EnableCaching
import org.springframework.cache.jcache.JCacheCacheManager
import org.springframework.context.annotation.Profile
import javax.cache.Caching


@Configuration
@Profile("!mock")
@EnableCaching
open class CacheConfig {
    @Bean
    open fun cacheManager(): CacheManager {
        val cachingProvider = Caching.getCachingProvider()
        val cacheManager = cachingProvider.getCacheManager(
                javaClass.getResource("/cache.xml").toURI(),
                cachingProvider.defaultClassLoader
        )
        return JCacheCacheManager(cacheManager)
    }
}